modules/queue/queue_disk.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package queue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+
+	"gitea.com/lunny/levelqueue"
+)
+
+// LevelQueueType is the type for level queue
+const LevelQueueType Type = "level"
+
+// LevelQueueConfiguration is the configuration for a LevelQueue
+type LevelQueueConfiguration struct {
+	DataDir      string
+	QueueLength  int
+	BatchLength  int
+	Workers      int
+	MaxWorkers   int
+	BlockTimeout time.Duration
+	BoostTimeout time.Duration
+	BoostWorkers int
+	Name         string
+}
+
+// LevelQueue implements a disk library queue
+type LevelQueue struct {
+	pool       *WorkerPool
+	queue      *levelqueue.Queue
+	closed     chan struct{}
+	terminated chan struct{}
+	exemplar   interface{}
+	workers    int
+	name       string
+}
+
+// NewLevelQueue creates a ledis local queue
```

Comment:
```suggestion
// NewLevelQueue creates a disk library queue
```

Reply:
So in this case I think the underlying implementation detail is important - the fact that this is a ledis queue shows that it's not clusterable and can't be read by more than one process at a time. If we move to a different disk implementation we can change it

Reply:
_"ledis queue"_? OMG, I thought it was a typo! 🤣

Reply:
This should be my typo on previous PR. It should be `leveldb based`.